spyder/plugins/maininterpreter.py

```diff
@@ -153,6 +158,28 @@ def python_executable_switched(self, custom):
             if custom:
                 self.warn_python_compatibility(cust_pyexec)
 
+    def is_python_interpreter(self, pyexec):
+        if not osp.isfile(pyexec):
+            return False
+        args = ["-c", "print('valid_python_interpreter')"]

```

Comment:
I think a better test is

1. Verify that the file is called `python`, `python2` or `python3`
2. Verify that it's a binary (using `is_text_file` from utils/encoding.py)
3. Run the command you defined here.

If those three criteria are met, then we can be more certain (although no completely sure :-) that the selected file is a Python interpreter.